src/transport/pci/bus.rs

```diff
@@ -695,3 +699,216 @@ impl From<u8> for HeaderType {
         }
     }
 }
+
+#[cfg(test)]
+mod tests {
+    use super::*;
+
+    #[test]
+    fn bar_info_unused() {
+        let device_function = DeviceFunction {
+            bus: 0,
+            device: 1,
+            function: 2,
+        };
+        let fake_cam = FakeCam {
+            device_function,
+            bar_values: [0, 0, 0, 0, 0, 0],
+            bar_masks: [
+                0xffffffff, 0xffffffff, 0xffffffff, 0xffffffff, 0xffffffff, 0xffffffff,
+            ],
+            status_command: 0,
+        };
+        let mut root = PciRoot::new(fake_cam);
+
+        assert_eq!(
+            root.bars(device_function).unwrap(),
+            [
+                Some(BarInfo::Memory {
```

Comment:
Maybe something for a separate API change, but it seems like it would be clearer to represent unimplemented BARs as `None` instead of `Some(..., size: 0, ...`.

Reply:
Is it possible to have unimplemented IO BARs or 64-bit memory BARs? The current API allows them to be distinguished, but I'm not sure whether that is actually relevant.

Reply:
An unimplemented base address register returns all 0 bits regardless of what is written to it, which just happens to correspond to the bit pattern for a memory-type BAR. So no, there is no "unimplemented IO BAR" or really "unimplemented <type> BAR" of any kind, and returning `BarInfo::Memory` for such a BAR is not really accurate. (Hopefully nobody is relying on the exact API for unused BARs anyway, but it would technically be a breaking API change.)

Reply:
Done in #192, PTAL.